lib/src/builder/parsing/version.dart

```diff
@@ -0,0 +1,240 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+
+import 'members.dart';
+
+/// The confidence levels related to boilerplate [Version]s.
+///
+/// As the builder detects members, it generates a confidence based on the
+/// implementation details of the member. This confidence score is then averaged
+/// over all of the grouped members (factory, props, state, component, etc).
+///
+/// Related: [resolveVersion]
+@sealed
+abstract class Confidence {
+  /// Not possible to be considered a given version.
+  static const none = 0;
+
+  /// Unlikely to be considered a given version.
+  ///
+  /// This value is an order of magnitude less than [neutral].
+  static const unlikely = 1;
+
+  /// Equally likely and unlikely to be considered a given version.
+  ///
+  /// Consuming code should only consider values strictly greater than this value
+  /// as confident enough to perform code generation:
+  ///
+  ///    - `confidence <= neutral`: do not perform code generation
+  ///    - `confidence > neutral`: perform code generation
+  ///
+  /// This value is an order of magnitude both greater than [unlikely] and less than [likely].
+  static const neutral = 10;
+
+  /// Likely to be considered a given version.
+  ///
+  /// This value is an order of magnitude both greater than [neutral] and less than [certain].
+  static const likely = 100;
+
+  /// Certain to be considered a given version.
+  ///
+  /// This value is an order of magnitude greater than [likely].
+  static const certain = 1000;
+
+  static const _confidenceToString = {
+    none: 'none',
+    unlikely: 'unlikely',
+    neutral: 'neutral',
+    likely: 'likely',
+    certain: 'certain',
+  };
+
+  /// Returns a string description of a [confidence] score, relating it to the constants in this
+  /// class.
+  ///
+  /// Example:
+  ///
+  ///     - Confidence.description(Confidence.neutral)    // 'neutral'
+  ///     - Confidence.description(Confidence.likely)     // 'likely'
+  ///     - Confidence.description(Confidence.likely + 1) // 'likely + 1'
+  static String description(int confidence) {
+    final baseConfidence = _confidenceToString.keys.where((c) => confidence >= c).reduce(max);
+    final difference = confidence - baseConfidence;
+
+    return '${_confidenceToString[baseConfidence]}${difference == 0 ? '' : ' + $difference'}';
+  }
+}
+
+/// The current versions of the boilerplate that the builder detects.
+///
+/// Related: [resolveVersion]
+enum Version {
+  // "v1" is for the transformer-based syntax :)
```

Comment:
What about pre-transformer? (codegen based) ;)

Reply:
Lolll I actually considered that, but then I was like "oh wait, that was before over_react was its own project"! 😄